### PR TITLE
[Python] Index querying pages batch [Replace C# samples]

### PR DESCRIPTION
**Pull request by reebhub**
[RDoc-2829](https://issues.hibernatingrhinos.com/issue/RDoc-2829/Python-Indexes-Querying-Querying-an-index-Replace-C-samples) Querying an index
[RDoc-2830](https://issues.hibernatingrhinos.com/issue/RDoc-2830/Python-ndexes-Querying-Filtering-Replace-C-samples) Filtering
[RDoc-2831](https://issues.hibernatingrhinos.com/issue/RDoc-2831/Python-ndexes-Querying-Exploration-queries-Replace-C-samples) Exploration queries
[RDoc-2832](https://issues.hibernatingrhinos.com/issue/RDoc-2832/Python-Indexes-Querying-Paging-Replace-C-samples) Paging
[RDoc-2833](https://issues.hibernatingrhinos.com/issue/RDoc-2833/Python-Indexes-Querying-Sorting-Replace-C-samples) Sorting
[RDoc-2834](https://issues.hibernatingrhinos.com/issue/RDoc-2834/Python-Indexes-Querying-Project-index-query-results-Replace-C-samples) Project index query results
[RDoc-2835](https://issues.hibernatingrhinos.com/issue/RDoc-2835/Python-Indexes-Querying-Full-Text-Search-Replace-C-samples) Full-Text Search
[RDoc-2836](https://issues.hibernatingrhinos.com/issue/RDoc-2836/Python-Indexes-Querying-Intersection-Replace-C-samples) Intersection
[RDoc-2837](https://issues.hibernatingrhinos.com/issue/RDoc-2837/Python-Indexes-Querying-Suggestions-Replace-C-samples) Suggestions
[RDoc-2838](https://issues.hibernatingrhinos.com/issue/RDoc-2838/Python-Indexes-Querying-Spatial-Replace-C-samples) Spatial
[RDoc-2839](https://issues.hibernatingrhinos.com/issue/RDoc-2839/Python-Indexes-Querying-Faceted-search-Replace-C-samples) Faceted search
[RDoc-2840](https://issues.hibernatingrhinos.com/issue/RDoc-2840/Python-Indexes-Querying-More-like-this-Replace-C-samples) More like this
[RDoc-2841](https://issues.hibernatingrhinos.com/issue/RDoc-2841/Python-Indexes-Querying-Highlighting-Replace-C-samples) Highlighting

### CODE REVIEW

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+
+{PANEL/}
+
+{PANEL: Syntax}
+
+{CODE:python syntax_1@Indexes\Querying\FacetedSearch.py /}
+
+| Parameter                    | Type                       | Description                                                                                       |
+|------------------------------|----------------------------|---------------------------------------------------------------------------------------------------|
+| **builder_or_facet** (Union) | `Callable[[FacetBuilder]`<br>**or**<br>`FacetBase` | Builder with a fluent API that constructs a `FacetBase` instance<br>**or**<br>`FacetBase` implementation defining the facet and its options |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:237]:
```suggestion
| **builder_or_facet** (Union) | `Callable[[FacetBuilder]`<br>**or**<br>`FacetBase` | Builder with a fluent API that constructs a `FacetBase` instance<br>**or**<br>`FacetBase` implementation instance|
```

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+| **facets**                   | `List[FacetBase]`                                  | A list of `FacetBase` implementations. |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:238]:
```suggestion
| **facets**                   | `List[FacetBase]`                                  | A list of `FacetBase` implementation instances. |
```

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown [hunk 3]
```diff
@@ -0,0 +1,279 @@
...
+{CODE:python syntax_6@Indexes\Querying\FacetedSearch.py /}
+
+| Parameter       | Type                        | Description |
+|-----------------|-----------------------------|-------------|
+| **field_name**  | `str` | The index-field to use for the facet |
+| **path**        | `str` | Points to the index-field to use for the facet (`ByRanges`, `ByField`) or for the aggregation (`SUM_ON`, `MIN_ON`, `MAX_ON`, `AVERAGE_ON`) |
+| **display_name** | `str` | If set, results of a facet will be returned under this name |
+| **options**     | `FacetOptions` | Non-default options to use in the facet definition |
+| **range_**     | `RangeBuilder` | A range of indexes |
+| **\*ranges**     | `RangeBuilder` | Multiple index ranges (at least one), separated by `,` |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:259]: the `range_` argument is the first one
if user would like to pass more, we have *ranges for it

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+{PANEL/}
+
+{PANEL: Options}
+
+Default parameters can be changed by manipulating `MoreLikeThisOptions` properties and passing them 
+to `more_like_this`.
+
+| Option | Type | Description |
+| ------------- | ------------- | ----- |
+| **MinimumTermFrequency** | `int` | Ignores terms with less than this frequency in the source doc |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown:66]: please `snake_case` the fields names

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+| **MaximumNumberOfTokensParsed** | `int` | The maximum number of tokens to parse in each example doc field that is not stored with TermVector support |
+| **MinimumWordLength** | `int` | Ignores words less than this length or, if 0, then this has no effect |
+| **MaximumWordLength** | `int` | Ignores words greater than this length or if 0 then this has no effect |
+| **MinimumDocumentFrequency** | `int` | Ignores words which do not occur in at least this many documents |
+| **MaximumDocumentFrequency** | `int` | Ignores words which occur in more than this many documents |
+| **MaximumDocumentFrequencyPercentage** | `int` | Ignores words which occur in more than this percentage of documents |
+| **Boost** | `bool` | Boost terms in query based on score |
+| **BoostFactor** | `float` |  Boost factor when boosting based on score |
+| **StopWordsDocumentId** | `str` | Document ID containing custom stop words |
+| **Fields** | `str[]` | Fields to compare |
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown:77]:
```suggestion
| **Fields** | `List[str]` | Fields to compare |

```

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/sorting.python.markdown [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+{CODE-TABS/}
+
+{INFO: }
+
+**Ordering Type**:
+
+* By default, the `order_by` methods will determine the `OrderingType` from the property path expression  
+  and specify that ordering type in the generated RQL that is sent to the server.
+
+* E.g. in the above example, ordering by `x => x.UnitsInStock` will result in `OrderingType.Int`
```
Comment (poissoncorp) [inline Documentation/5.4/Raven.Documentation.Pages/indexes/querying/sorting.python.markdown:44]:
```suggestion
* E.g. in the above example, ordering by `UnitsInStock` will result in `OrderingType.Int`  
```

**Review by poissoncorp — CHANGES_REQUESTED** [review]
let's fix these small issues & we good to go :]

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown (hunk 1 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:237]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown (hunk 2 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:238]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown (hunk 3 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/faceted-search.python.markdown:259]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown (hunk 4 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown:66]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown (hunk 5 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/morelikethis.python.markdown:77]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/projections.python.markdown [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+     Name : e.FirstName, 
+     Metadata : getMetadata(e)
+}
+{CODE-TAB-BLOCK/}
+{CODE-TABS/}
+
+{PANEL/}
+
+{PANEL: Projection Behavior}
+The `aelect_fields` methods can also take a `ProjectionBehavior` parameter, which
```
[the remark this answers is not in the document]
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/projections.python.markdown:288]: done

File: Documentation/5.4/Raven.Documentation.Pages/indexes/querying/sorting.python.markdown (hunk 6 above)
Comment (reebhub) [reply Documentation/5.4/Raven.Documentation.Pages/indexes/querying/sorting.python.markdown:44]: done

**Pull request merged**